src/sentry/integrations/bitbucket_server/repository.py

```diff
@@ -112,18 +114,53 @@ def compare_commits(self, repo, start_sha, end_sha):
         client = installation.get_client()
         # use config name because that is kept in sync via webhooks
 
-        # Bitbucket servers send an empty commit as a string of zeros
-        if "0" * 40 == start_sha:
-            start_sha = None
-
-        try:
-            res = client.get_commits(
-                repo.config["project"], repo.config["repo"], start_sha, end_sha
-            )
-        except Exception as e:
-            installation.raise_error(e)
+        if "0" * 40 == start_sha or start_sha is None:
+            try:
+                res = client.get_last_commits(repo.config["project"], repo.config["repo"], 10)
```

Comment:
@pierredup You could probably make this more concise if you put the if/else block inside the try catch and not using that last else block after the exception:
```
try:
    if "0" * 40 == start_sha or start_sha is None:
        res = client.get_last_commits(repo.config["project"], repo.config["repo"], 10)
    else:
        res = client.get_commits( repo.config["project"], repo.config["repo"], start_sha, end_sha)
    return self._format_commits(client, repo, res)
except Exception as e:
    installation.raise_error(e)   
```
Though I'm not 100% sure that code is error free.

Reply:
I've made the change but haven't tested it locally. Will do a full test again in the next couple of days